File: DouYu/app/src/test/java/cn/edu/app/douyu/network/ApiResponseTest.java
package cn.edu.app.douyu.network;

import org.junit.Test;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

public class ApiResponseTest {
    @Test
    public void acceptsBackendOkAndSuccessCodes() {
        ApiResponse<String> ok = new ApiResponse<>();
        ok.code = "OK";
        assertTrue(ok.success());

        ApiResponse<String> success = new ApiResponse<>();
        success.code = "SUCCESS";
        assertTrue(success.success());
    }

    @Test
    public void rejectsNonSuccessCodes() {
        ApiResponse<String> response = new ApiResponse<>();
        response.code = "UNAUTHORIZED";
        assertFalse(response.success());
    }
}

File: DouYu/app/src/test/java/cn/edu/app/douyu/core/ApiResponseTest.kt
package cn.edu.app.douyu.core

import cn.edu.app.douyu.core.network.ApiResponse
import org.junit.Assert.assertEquals
import org.junit.Assert.assertTrue
import org.junit.Test

class ApiResponseTest {
    @Test
    fun okResponseCarriesTraceAndData() {
        val response = ApiResponse(
            code = "OK",
            message = "success",
            data = "pong",
            traceId = "trace_001"
        )

        assertTrue(response.isOk)
        assertEquals("pong", response.data)
        assertEquals("trace_001", response.traceId)
    }
}
